Zend/zend_execute.c

```diff
@@ -1795,6 +1795,18 @@ static zend_never_inline void zend_fetch_dimension_address_UNSET(zval *result, z
 	zend_fetch_dimension_address(result, container_ptr, dim, dim_type, BP_VAR_UNSET);
 }
 
+static zend_always_inline int zend_fetch_dimension_address_read_valid_container(zval *container, int support_strings) {
+	if (EG(current_execute_data)->opline->opcode != ZEND_FETCH_LIST) {
+		if (EG(current_execute_data)->opline->op1_type != IS_VAR) {
+			return 0;
+		} else if (Z_TYPE_P(container) != IS_UNDEF && Z_TYPE_P(container) != IS_NULL && support_strings && Z_TYPE_P(container) != IS_STRING) {
```

Comment:
IS_UNDEF and IS_STRING are already checked before.
So the check for IS_NULL should be enough.